File: javascript/leetcode/contains-duplicate-ii.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function(nums, k) {
    seen = {}
    for (var i = 0; i < nums.length; i++) {
        if (seen[nums[i]] !== null && i - seen[nums[i]] <= k) { return true }
        seen[nums[i]] = i
    }
    return false
};

File: java/leetcode/contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        HashMap<Integer, Integer> seen = new HashMap<Integer, Integer>();
        int seenValue;
        for (int i = 0; i < nums.length; i++) {
            if (seen.containsKey(nums[i]) && i - seen.get(nums[i]) <= k) { return true; }
            seen.put(nums[i], i);
        }
        return false;
    }
}
